framework/src/wifi_manager/wifi_utils_lwnl80211.c

```diff
@@ -355,11 +355,17 @@ wifi_utils_result_e wifi_utils_deinit(void)
 wifi_utils_result_e wifi_utils_scan_ap(void *arg)
 {
 	WU_ENTER;
+	wifi_utils_ap_config_s *config = NULL;
+	uint32_t config_len = 0;
+	if (arg) {
+		config = (wifi_utils_ap_config_s *)arg;
```

Comment:
lwnl80211_ap_config_s is defined in lwnl80211.h(os/include) so framework can access it.
I think you'd better use it directly instead of wifi_utils_ap_config_s or use lwnl80211_ap_config_s as a subset.
Then you don't have to use type casting and reduce size of argument which being send through the ioctl.
Also It will prevent conflict between framework and driver.

Reply:
Thanks for the comment. This is really sharp comment. Actually, lwnl80211 is a kind of driver to separate the user space from the kernel in WiFi stack. In case of flat build environment, we do not need to use lwnl80211, but wifi_utils layer is still required as a porting layer. Based on this layering, wifi_utils_ap_config_s is a common structure, independent from the usage of lwnl80211. That is why we still use wifi_utils_ap_config_s instead of your suggestion. For the next, we will remove wifi_utils porting layer in the end, at that time I will remove wifi_utils_ap_config_s, adopting lwnl80211_ap_config_s as your suggestion. Thanks!